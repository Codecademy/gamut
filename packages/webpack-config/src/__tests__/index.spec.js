const { createConfig } = require('../index');

describe('createConfig', () => {
  it('merges custom config', () => {
    const testConfig = createConfig()
      .common({
        context: __dirname,
      })
      .merge({
        entry: 'my-file.js',
      })
      .toConfig();

    expect(testConfig.entry).toEqual('my-file.js');
  });

  it('merges loaders', () => {
    const testConfig = createConfig()
      .common({
        context: __dirname,
      })
      .if(false, config =>
        config.merge({
          notInTheSnapshot: true,
        })
      )
      .mergeLoader({
        test: /\.js?$/,
        loader: 'babel-loader',
        include: [__dirname],
      })
      .toConfig();

    expect(testConfig).toMatchSnapshot();
  });
});
