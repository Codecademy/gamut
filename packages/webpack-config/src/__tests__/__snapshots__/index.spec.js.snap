// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createConfig merges loaders 1`] = `
Object {
  "context": "/Users/jake/Sites/Codecademy/npm/webpack-config/src/__tests__",
  "entry": "/Users/jake/Sites/Codecademy/npm/webpack-config/src/__tests__/src/main.js",
  "module": Object {
    "rules": Array [
      Object {
        "include": Array [
          "/Users/jake/Sites/Codecademy/npm/webpack-config/src/__tests__",
        ],
        "loader": "babel-loader",
        "options": Object {
          "cacheDirectory": true,
          "presets": Array [
            "codecademy",
          ],
        },
        "test": /\\\\\\.js\\?\\$/,
      },
      Object {
        "loader": "file-loader",
        "test": /\\\\\\.jpe\\?g\\$\\|\\\\\\.gif\\$\\|\\\\\\.png\\$\\|\\\\\\.svg\\$\\|\\\\\\.woff\\$\\|\\\\\\.ttf\\$\\|\\\\\\.eot\\|\\\\\\.wav\\$\\|\\\\\\.mp3\\$/,
      },
    ],
  },
  "output": Object {
    "filename": "[name].js",
    "path": "/Users/jake/Sites/Codecademy/npm/webpack-config/src/__tests__/dist",
  },
  "plugins": Array [
    DefinePlugin {
      "definitions": Object {
        "__DEV__": true,
        "process.env.NODE_ENV": "\\"development\\"",
      },
    },
    ContextReplacementPlugin {
      "newContentRecursive": undefined,
      "newContentRegExp": /en/,
      "newContentResource": undefined,
      "resourceRegExp": /moment\\[\\\\/\\\\\\\\\\]locale\\$/,
    },
  ],
  "resolve": Object {
    "extensions": Array [
      ".webpack.js",
      ".web.js",
      ".js",
      ".scss",
      ".css",
      ".json",
    ],
    "modules": Array [
      "node_modules",
      "node_modules/@codecademy",
    ],
  },
}
`;
